nikola/plugins/compile/markdown/__init__.py

```diff
@@ -49,35 +49,38 @@ class CompileMarkdown(PageCompiler):
     name = "markdown"
     friendly_name = "Markdown"
     demote_headers = True
-    extensions = []
+    markdown = None
     site = None
 
     def set_site(self, site):
         """Set Nikola site."""
         super(CompileMarkdown, self).set_site(site)
         self.config_dependencies = []
+        extensions = []
         for plugin_info in self.get_compiler_extensions():
             self.config_dependencies.append(plugin_info.name)
-            self.extensions.append(plugin_info.plugin_object)
+            extensions.append(plugin_info.plugin_object)
             plugin_info.plugin_object.short_help = plugin_info.description
 
         site_extensions = self.site.config.get("MARKDOWN_EXTENSIONS")
         self.config_dependencies.append(str(sorted(site_extensions)))
-        self.extensions.extend(site_extensions)
+        extensions.extend(site_extensions)
+        self.markdown = Markdown(extensions=extensions, output_format="html5")
```

Comment:
Are `Markdown` objects thread-safe? (You can tell doit to process tasks in parallel using threads, so it could be that two Markdown posts are compiled simultaneously.) I somehow doubt this as you call `self.markdown.reset()` after `self.markdown.convert(data)`. If this is the case, you should better use thread-local storage for the `Markdown` object.
(That would also indirectly solve Kwpolska's remark.)